dask/blockwise.py

```diff
@@ -617,7 +617,15 @@ def clone(
 
         indices = []
         for k, idxv in self.indices:
-            if idxv is not None and k in names:
+            try:
+                in_names = k in names
+            except TypeError:
```

Comment:
This seems reasonable - I suppose you can also use `if dask.core.ishashable(k) and k in names` if that feels any clearer to you.

Reply:
good suggestion; implemented